mllib/src/main/scala/org/apache/spark/ml/feature/QuantileDiscretizer.scala

```diff
@@ -49,6 +49,20 @@ private[feature] trait QuantileDiscretizerBase extends Params
 
   /** @group getParam */
   def getNumBuckets: Int = getOrDefault(numBuckets)
+
+  /**
+   * Relative error (see approxQuantile documentation for description).  Must be >= 0.
+   * default: 0.01
+   * @group param
+   */
+  val relativeError = new DoubleParam(this, "relativeError", "The relative target precision " +
+    "for approxQuantile",
+    ParamValidators.gtEq(0.0))
+  setDefault(relativeError -> 0.01)
+
+  /** @group Param */
+  def getRelativeError: Double = get(relativeError).getOrElse
+    { math.min(0.01, 1.0 / (10.0 * getNumBuckets)) }
```

Comment:
From a quick look at the paper, it does seem like a default range of `0.001 - 0.01` for eps is reasonable.